Magecloud 2903 update php version validation

meker12 opened the pull request:
## This PR is a:

- [ ] New topic
- [x] Content update
- [ ] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary
- Added environment configuration validation to check the PHP version and warn users if they are not using the latest recommended version.
- Updated the Magento Commerce Cloud Technologies and requirements topic to include links to the Magento technology stack requirements for each version.

**Affected URLs**

*Technologies and requirements*
- https://devdocs.magento.com/guides/v2.1/cloud/requirements/cloud-requirements.html
- https://devdocs.magento.com/guides/v2.2/cloud/requirements/cloud-requirements.html
- https://devdocs.magento.com/guides/v2.3/cloud/requirements/cloud-requirements.html
*Cloud release notes*
- https://guides/v2.1/cloud/release-notes/cloud-tools.html
- https://guides/v2.2/cloud/release-notes/cloud-tools.html





List all affected URLs 

- ...
- ...

<!-- (REQUIRED) The Url that this PR will modify -->

<!-- (OPTIONAL) What other information can you provide about this PR? -->

<!--
Thank you for your contribution!

Before submitting this pull request, please make sure you have read our Contribution Guidelines and your PR meets our contribution standards:
https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md

Please fill out as much information as you can about your PR to help speed up the review process.
If your PR addresses an existing GitHub Issue, please refer to it in the title or Additional Information section to make the connection.

We may ask you for changes in your PR in order to meet the standards set in our Contribution Guidelines. PR's that do not comply with our guidelines may be closed at the maintainers' discretion.

Feel free to remove this section before creating this PR.
-->

hguthrie commented on guides/v2.1/cloud/requirements/cloud-requirements.md:
```diff
@@ -72,17 +72,20 @@ You must use Secure Shell (SSH) and not HTTPS to connect to the Git repository.
 
-For Staging and Production, you will use Fastly for CDN and caching. We recommend installing Fastly module 1.2.33 or later. For details, see [Fastly in Cloud]({{ page.baseurl }}/cloud/basic-information/cloud-fastly.html).
+{:.bs-callout .bs-callout-info}
+See [Magento 2.3.x technology stack requirements]({{ page.baseurl }}/install-gde/system-requirements-tech.html) for the latest software version compatibility information. 
```
yay!

hguthrie commented on guides/v2.1/cloud/requirements/cloud-requirements.md:
```diff
@@ -72,17 +72,20 @@ You must use Secure Shell (SSH) and not HTTPS to connect to the Git repository.
-For detailed information on supported versions and extensions, see the following information. These files allow you to configure software versions you want to use in your implementation.
+For Staging and Production environments, you use the Fastly CDN for Magento2 extension for CDN and caching services. See [Configure Fastly services]({{ page.baseurl }}/cloud/cdn/cloud-fastly.html#fastly-cdn-module-for-magento).
+
+For information about configuring the software versions to use in your implementation, see the documentation for the following {{ site.data.var.ece }} configuration files:
```
"see the documentation" seems weird

meker12 commented:
> Looks like something weird with the 2.3 Technologies and req file. Deleted, added again, and can't comment because of a symlink problem.

-- The build looks OK.  Each version has its own requirements file now-- with links to the MC technology stack requirements for associated version.
